src/algorithm/crba.hxx

```diff
@@ -161,27 +161,41 @@ namespace se3
                      const se3::Model & model,
                      se3::Data & data)
     {
-      typedef typename SizeDepType<JointModel::NV>::template ColsReturn<Matrix6x>::Type ColsBlock;
+      typedef typename SizeDepType<JointModel::NV>::template ColsReturn<Data::Matrix6x>::Type ColsBlock;
       
       const Model::JointIndex & i = (Model::JointIndex) jmodel.id();
       const Model::Index & parent = model.parents[i];
       
       data.Ycrb[parent] += data.liMi[i].act(data.Ycrb[i]);
       
       jdata.U() = data.Ycrb[i] * jdata.S();
-      if (JointModel::NV == -1)
-      {
-        ColsBlock jF
-        = data.Ag.middleCols(jmodel.idx_v(), jmodel.nv());
-      }
-      else
-      {
-        ColsBlock jF
+
+      ColsBlock jF
         = data.Ag.middleCols <JointModel::NV> (jmodel.idx_v());
-      }
+
       forceSet::se3Action(data.oMi[i],jdata.U(),jF);
     }
     
+    static void algo(const se3::JointModelBase<JointModelComposite> & jmodel,
+                     se3::JointDataBase<JointDataComposite> & jdata,
+                     const se3::Model & model,
+                     se3::Data & data)
+    {
+      typedef SizeDepType<JointModel::NV>::ColsReturn<Data::Matrix6x>::Type ColsBlock;
+      
+      const Model::JointIndex & i = (Model::JointIndex) jmodel.id();
+      const Model::Index & parent = model.parents[i];
+      
+      data.Ycrb[parent] += data.liMi[i].act(data.Ycrb[i]);
+      
+      jdata.U() = data.Ycrb[i] * jdata.S();
+      
+      ColsBlock jF
```

Comment:
The sizes NQ, NV for a JointComposite are not known at compilation as it can hold any type of joint. Hence these values are set to Eigen::Dynamic ( that's why there was previous tests like ' if NV == -1'.
This algo has been specialized specificly  for Jointcomposite in order not to loose the templated version of middleCols for other type of joints. I agree that duplicating the code for only this line is surely too much (I didn't succeed to have the 'ColsBlock jF' line compiling and properly set for both cases of JointComposite or other joints in only one function. One easy solution would be to use the dynamic version of middleCols for both cases. Should i change to this ?